tests/dummy/app/components/code-template-toggle/component.js

```diff
@@ -10,40 +9,45 @@ export default class CodeTemplateToggleComponent extends Component {
   showCode = true;
   _toggleTimer = null;
 
+  maxHeight = 0;
+
   didInsertElement() {
     super.didInsertElement(...arguments);
-    this._toggleTimer = schedule('afterRender', null, () => {
-      const element = document.getElementById(this.id);
-
-      if (!element) {
-        return;
-      }
-
-      let sectionToggles = element.querySelectorAll(
-        '.code-template-toggle-section'
-      );
-      let maxHeight = Math.max.apply(
-        null,
-        [].map.call(sectionToggles, function (el) {
-          return el.offsetHeight;
-        })
-      );
-
-      let toggle = element.querySelector('.code-template-toggle');
+    const element = document.getElementById(this.id);
+    if (!element) {
+      return;
+    }
 
+    let resizeObserver = this._createResizeObserver((maxHeight) => {
```

Comment:
do we need to call `disconnect()` on the observer in `willDestroyElement`?

Reply:
Good point! Updated it with the cleanup in `willDestroyElement`